src/components/marker-chart/Canvas.js

```diff
@@ -97,40 +98,66 @@ class MarkerChartCanvas extends React.PureComponent<Props, State> {
       Math.ceil(viewportBottom / rowHeight),
       markerTimingAndBuckets.length
     );
-
+    const markerIndexToTimingRow = this._getMarkerIndexToTimingRow(
+      markerTimingAndBuckets
+    );
+    const rightClickedRow: number | void =
+      rightClickedMarkerIndex === null
+        ? undefined
+        : markerIndexToTimingRow.get(rightClickedMarkerIndex);
+    const newRow: number | void =
+      hoveredItem === null
+        ? undefined
+        : markerIndexToTimingRow.get(hoveredItem);
+        
     if (isHoveredOnlyDifferent) {
       // Only re-draw the rows that have been updated if only the hovering information
       // is different.
-      const markerIndexToTimingRow = this._getMarkerIndexToTimingRow(
-        markerTimingAndBuckets
-      );
 
       const oldRow: number | void =
         prevHoveredItem === null
           ? undefined
           : markerIndexToTimingRow.get(prevHoveredItem);
-      const newRow: number | void =
-        hoveredItem === null
-          ? undefined
-          : markerIndexToTimingRow.get(hoveredItem);
 
       if (newRow !== undefined) {
         this.clearRow(ctx, newRow);
+        this.highlightRow(ctx, newRow);
         this.drawMarkers(ctx, hoveredItem, newRow, newRow + 1);
       }
       if (oldRow !== undefined && oldRow !== newRow) {
-        this.clearRow(ctx, oldRow);
+        if (oldRow !== rightClickedRow) {
+          this.clearRow(ctx, oldRow);
+        }
         this.drawMarkers(ctx, hoveredItem, oldRow, oldRow + 1);
       }
     } else {
       ctx.fillStyle = '#ffffff';
       ctx.fillRect(0, 0, containerWidth, containerHeight);
-
+      if (rightClickedRow !== undefined) {
+        this.highlightRow(ctx, rightClickedRow);
+      } else if (newRow !== undefined) {
```

Comment:
So without this `else if` condition when we right-clicked on a marker and then left clicked on a different marker on a different row then the row was not highlighted.

Reply:
yes this sounds right!